client/components/CreateItem.js

```diff
@@ -0,0 +1,122 @@
+import React from "react";
+import { connect } from "react-redux";
+import { createItem } from "../store/itemsReducer";
+import { fetchUser } from "../store/userReducer";
+
+class CreateItem extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      name: "",
+      description: "",
+      price: 0,
+      inventory: 100,
+      rarity: 1,
+      itemType: "",
+      // type: ('sword', 'dagger', 'bow', 'magic'),
+      itemClass: "",
+      // type: ('adventurer', 'villain'),
+      imageUrl: "",
+    };
+    this.save = this.save.bind(this);
+  }
+  componentDidMount() {
+    this.props.validateUser();
+  }
+  save(ev) {
+    ev.preventDefault();
+    this.props.createItem(
+      {
+        name: this.state.name,
+        description: this.state.description,
+        price: Number(this.state.price),
+        imageUrl: this.state.imageUrl,
+        inventory: Number(this.state.inventory),
+        rarity: Number(this.state.rarity),
+        itemType: this.state.itemType,
+        itemClass: this.itemClass,
+      },
+      this.props.history
+    );
+  }
+  render() {
+    const isAdmin = this.props.user && this.props.user.class === "admin";
+    const {
+      name,
+      description,
+      price,
+      imageUrl,
+      inventory,
+      rarity,
+      itemType,
+      itemClass,
+    } = this.state;
+    const { save } = this;
+    return isAdmin ? (
+      <div>
+        <form onSubmit={save}>
+          <div>Item Name:</div>
+          <input
+            value={name}
+            onChange={(ev) => this.setState({ name: ev.target.value })}
+          />
+          <div>Item Description: </div>
+          <input
+            value={description}
+            onChange={(ev) => this.setState({ description: ev.target.value })}
+          />
+          <div>Item Price: $</div>
+          <input
+            value={price}
+            onChange={(ev) => this.setState({ price: ev.target.value })}
+          />
+          <div>Item imageUrl: </div>
+          <input
+            value={imageUrl}
+            onChange={(ev) => this.setState({ imageUrl: ev.target.value })}
+          />
+          <div>Item Inventory: </div>
+          <input
+            value={inventory}
+            onChange={(ev) => this.setState({ inventory: ev.target.value })}
+          />
+          <div>Item Rarity: </div>
+          <input
+            value={rarity}
+            onChange={(ev) => this.setState({ rarity: ev.target.value })}
```

Comment:
maybe it'd be better to use a select/option (dropdown) menu with this? that way, you can have it actually choose "rare, epic, unique, legendary" items in the dropdown